src/util/screensaver.cpp

```diff
@@ -183,100 +209,119 @@ void ScreenSaverHelper::triggerUserActivity()
+            s_enabled = true;
+            qDebug() << "Idle inhibitor" << kFreeDesktopPortal.interface << "enabled";
+            return;
+        } else {
+            qWarning() << "Inhibit for" << kFreeDesktopPortal.interface << "failed: "
+                       << reply.error().message();
+        }
+    } else {
+        qDebug() << "Interface" << kFreeDesktopPortal.interface << "is not valid";
+    }
+
+    for (int i = 0; i < kDBusInhibitors.size(); ++i) {
+        // Portal unavailable, try to use standard D-Bus idle inhibit interfaces
+        QDBusInterface iface(kDBusInhibitors[i].service,
+                kDBusInhibitors[i].object_path,
+                kDBusInhibitors[i].interface,
+                QDBusConnection::sessionBus());
         if (iface.isValid()) {
-            QDBusReply<void> reply = iface.call(USERACTIVITY[i][3]);
+            QDBusReply<uint32_t> reply = iface.call("Inhibit",
+                    "org.mixxx.Mixxx",
+                    "Mixxx is running");
             if (reply.isValid()) {
-                s_sendActivity = true;
-                break;
+                inhibitor = &kDBusInhibitors[i];
+                inhibit_cookie = reply.value();
+                s_enabled = true;
+                qDebug() << "Idle inhibitor" << kDBusInhibitors[i].interface << "enabled";
+                return;
             } else {
-                errors = errors +  "\nCall to inhibit for " + USERACTIVITY[i][0] + " failed: " 
-                    + reply.error().message();
+                qWarning() << "Inhibit for"
+                           << kDBusInhibitors[i].interface << "failed: "
+                           << reply.error().message();
             }
-        } 
+        } else {
+            qDebug() << "Interface" << kDBusInhibitors[i].interface << "is not valid";
+        }
     }
-    if (!s_sendActivity) {
-        qWarning() << "Could not send activity using the registered DBus methods. " 
-        << "Errors were: " << errors << 
-        "\nWill try to use the Xlib XResetScreensaver instead.";
+
+    if (!inhibitor) {
+        qDebug() << "No supported idle inhibitation services found";
     }
 }
-#endif // 0
 
-void ScreenSaverHelper::inhibitInternal()
-{
-    if (!QDBusConnection::sessionBus().isConnected()) {
-        qWarning("Cannot connect to the D-Bus session bus.\nTo start it, run:\n"
-                "\teval `dbus-launch --auto-syntax`");
+void ScreenSaverHelper::uninhibitInternal() {
+    if (!inhibitor) {
+        qDebug() << "Cannot uninhibit without a selected interface";
         return;
     }
-    if (s_cookie > 0) {
-        uninhibit();
-    }
-    s_cookie = 0;
-    for (int i=0; SCREENSAVERS[i][0] != nullptr; i++ ) {
-        QDBusInterface iface(SCREENSAVERS[i][0], SCREENSAVERS[i][1], SCREENSAVERS[i][2], 
-            QDBusConnection::sessionBus());
+
+    if (inhibitor->service == "org.freedesktop.portal.Desktop" && !session_handle.isEmpty()) {
```

Comment:
wouldn't this also work?
```suggestion
    if (inhibitor == &kFreeDesktopPortal && !session_handle.isEmpty()) {
```